### PR TITLE
Interface modifiers and other fixes

### LINKED ISSUE #627: multi-line definition with string interpolation for local_screen
First check out our main documentation site at [https://openc3.com](https://openc3.com).

**Describe the bug**
The string interpolation requires the use of double-quote. In the example single quote is used and works as expected.

Introducing string interpolation and multi-line causes some instrumented code to inject itself into the string causing a bad screen definition. The \ at the end of each line worked for most lines until #{} was used.

In an attempt to find a work around, the only solution I could come up with was using \n and not using multi-line formatting. Example:
definition = "SCREEN AUTO AUTO 1.0 \n VERTICALBOX 'Test Screen' \n LABELVALUE #{target} #{packet} #{item} \n  END"
local_screen("TEST", definition)

**To Reproduce**
Create a local screen definition with string interpolation and multi-line formatting
```
target = "INST"
packet = "HEALTH_STATUS"
item = "TEMP1"
definition = "
  SCREEN AUTO AUTO 1.0 
    VERTICALBOX 'Test Screen' 
    LABELVALUE #{target} #{packet} #{item} 
  END"
local_screen("TEST", definition)
```
Then select the edit button to show the screen definition. You should see instrumented code injected. This causes issues if you try to add more than 1 value to the screen.

**Expected behavior**
The definition's string should be the only injected string into the screens definition.

**Environment (please complete the following information):**

- OS: Windows 10
- OpenC3 COSMOS Version 5.6.0
- Browser
  -- [] Internet Explorer (IE)
  -- [x] Google Chrome
  -- [] Chromium based browsers
  -- [] Mozilla Firefox
  -- [] Safari

### LINKED ISSUE #653: Add PORT and ROUTER_PREFIX settings to interface and routers
PORT used for configuring a service in Kubernetes
ROUTER_PREFIX used to expose a http server

### PR DESCRIPTION
**Pull request by ryanmelt**
closes #653 
closes #627

### CODE REVIEW

File: openc3/lib/openc3/utilities/ruby_lex_utils.rb [hunk 1]
```diff
@@ -292,32 +292,9 @@ def each_lexed_segment(text)
...
         elsif !line.empty?
-          num_left_brackets  = line.count('{')
-          num_right_brackets = line.count('}')
-          if num_left_brackets != num_right_brackets
-            # Don't instrument lines with unequal numbers of { and } brackets
-            yield line, false, inside_begin, lex.exp_line_no
-          else
-            yield line, true, inside_begin, lex.exp_line_no
-          end
+          yield line, true, inside_begin, lex.exp_line_no
```
Comment (jmthomas) [inline openc3/lib/openc3/utilities/ruby_lex_utils.rb:297]: What were you trying to accomplish with these two blocks that got removed?

File: openc3/lib/openc3/packets/parsers/xtce_parser.rb [hunk 2]
```diff
@@ -699,6 +699,8 @@ def set_min_max_default(item, type, data_type)
               # Strip quotes from strings
               if type.initialValue[0] == '"' && type.initialValue[-1] == '"'
                 item.default = type.initialValue[1..-2]
+              else
+                item.default = type.initialValue
```
Comment (jmthomas) [inline openc3/lib/openc3/packets/parsers/xtce_parser.rb:703]: Are quote optional with XTCE?

File: openc3/lib/openc3/io/json_api.rb [hunk 3]
```diff
@@ -65,7 +65,7 @@ def _generate_auth
 
     def _request(*method_params, **kw_params)
       kw_params[:scope] = $openc3_scope unless kw_params[:scope]
-      kw_params[:json] => true unless kw_params[:json]
+      kw_params[:json] = true unless kw_params[:json]
```
Comment (jmthomas) [inline openc3/lib/openc3/io/json_api.rb:68]: This previously looks like a syntax error. Not even sure what this does. Did you look for other examples?

**Review by jmthomas — APPROVED** [review]

File: openc3/lib/openc3/utilities/ruby_lex_utils.rb (hunk 1 above)
Comment (ryanmelt) [reply openc3/lib/openc3/utilities/ruby_lex_utils.rb:297]: They handled cases that now appear to be handled by paying attention to ltype.

File: openc3/lib/openc3/packets/parsers/xtce_parser.rb (hunk 2 above)
Comment (ryanmelt) [reply openc3/lib/openc3/packets/parsers/xtce_parser.rb:703]: Apparently. Someone ran into this case in COSMOS 4.

File: openc3/lib/openc3/io/json_api.rb (hunk 3 above)
Comment (ryanmelt) [reply openc3/lib/openc3/io/json_api.rb:68]: It was a bug for sure.  Setting the json flag adds the appropriate content-type header.

**Pull request merged**